components/speedreader/speedreader_service.cc

```diff
@@ -128,20 +129,27 @@ void SpeedreaderService::IncrementPromptCount() {
   prefs_->SetInteger(kSpeedreaderPrefPromptCount, count + 1);
 }
 
-void SpeedreaderService::SelectedTheme(const std::string& theme_name) {
-#if DCHECK_IS_ON()
-  static constexpr const char* kValidNames[] = {
-      "",
-      "system",  // same as ''
-      "light",  "sepia", "dark",
-  };
-  DCHECK(base::Contains(kValidNames, theme_name));
-#endif
-  prefs_->SetString(kSpeedreaderPrefThemeName, theme_name);
+void SpeedreaderService::SetTheme(Theme theme) {
+  prefs_->SetInteger(kSpeedreaderPrefTheme, static_cast<int>(theme));
 }
 
-std::string SpeedreaderService::GetSelectedTheme() const {
-  return prefs_->GetString(kSpeedreaderPrefThemeName);
+Theme SpeedreaderService::GetTheme() const {
+  return static_cast<Theme>(prefs_->GetInteger(kSpeedreaderPrefTheme));
+}
+
+std::string SpeedreaderService::GetThemeName() const {
+  switch (GetTheme()) {
+    case Theme::kDefault:
+      return {};
+    case Theme::kLight:
+      return "light";
+    case Theme::kSepia:
+      return "sepia";
+    case Theme::kDark:
+      return "dark";
+  }
+  NOTREACHED();
```

Comment:
why `NOTREACHED` and not a `default:` switch case? a default switch case will always fallback

Reply:
```c++
switch (s) {
  // cases
  default: return {};
}
// nothing needed here 
```